src/components/AllArticles.jsx

```diff
@@ -13,12 +13,12 @@ function AllArticles() {
       .then((articles) => {
         setArticles(articles);
       })
-      .then(() => {
+      .finally(() => {
```

Comment:
So there are a lot of interesting things here going on.
First of all on line 9, where you are creating your isLoading state, you initially have the state set to true. I'd argue it would make more sense to set it to false, set it to true in the useEffect before the fetchArticles, and then again set it to false AFTER the articles have been set, similarly to how you're doing here.

I would strongly encourage you to use a .catch, and instead of using a .finally use a .then.
Although they do similar things it's important to note that a .finally will always run, _no matter if there has been an error or not_

So in the worst case scenario in case of there being an error, your users would either have a page showing nothing or a crashed page. Not ideal when we want to give users some sort of feedback!